enterprise/app/workflows/workflows.tsx

```diff
@@ -415,6 +418,21 @@ class RepoItem extends React.Component<RepoItemProps, RepoItemState> {
       .executeWorkflow(
         new workflow.ExecuteWorkflowRequest({
           pushedRepoUrl: this.props.repoUrl,
+          // Parse "var=val" pairs from the input field to Record<string, string>
+          env: this.state.runWorkflowActionNames
+            .split(",")
+            .filter((n) => n.includes("="))
```

Comment:
Rather than just filtering out invalid input, maybe we should return an error message and some additional validation that the input looks correct

Reply:
I was hoping to get away with a simple implementation first and add UI improvements to this later.

I expect the current `Env` field to be sort-lived, and that we should add a table input (2 columns) for users to add the environment variables in later. That would save us from this whole validation madness. Wdyt?

Reply:
```diff
--- a/enterprise/app/workflows/workflows.tsx
+++ b/enterprise/app/workflows/workflows.tsx
@@ -414,21 +414,28 @@ class RepoItem extends React.Component<RepoItemProps, RepoItemState> {
     this.setState({ runWorkflowActionStatuses: null, isWorkflowRunning: true });
     this.onCloseMenu();

+    // Parse "var1=val1,var2=val2" string from the input field to Record<string, string>
+    const env: Record<string, string> = {};
+    const envPairs = this.state.runWorkflowEnv.split(",")
+    if (!envPairs.reduce((acc, pair) => {
+      if (!acc) return false;
+      const halves = pair.split("=");
+      if (halves.length !== 2) return false;
+      // We just care about variable not being empty, value can be empty
+      if (halves[0].trim().length === 0) return false;
+
+      env[halves[0].trim()] = halves[1].trim();
+      return true;
+    }, true)) {
+      errorService.handleError("Invalid environment variable format. Please use 'VAR1=value1,VAR2=value2'.");
+      return;
+    }
+
     rpcService.service
       .executeWorkflow(
         new workflow.ExecuteWorkflowRequest({
           pushedRepoUrl: this.props.repoUrl,
-          // Parse "var1=val1,var2=val2" string from the input field to Record<string, string>
-          env: this.state.runWorkflowEnv
-            .split(",")
-            .filter((n) => n.includes("="))
-            .map((n) => n.trim().split("="))
-            .filter(([k, v]) => k && v)
-            .map(([k, v]) => [k.trim(), v.trim()])
-            .reduce((acc: Record<string, string>, [k, v]) => {
-              acc[k] = v;
-              return acc;
-            }, {}),
+          env: env,
           actionNames: this.state.runWorkflowActionNames
             .split(",")
             .map((n) => n.trim())
```
fwiw, here is what I cooked up locally which is really ugly 😆